spec/support/active_model_serializers_helpers.rb

```diff
@@ -0,0 +1,54 @@
+require 'active_model_serializers'
+
+module ActiveModelSerializersHelpers
```

Comment:
Can we move this into the `spec/ddtrace/contrib/active_model_serializers` folder, since these helpers are specific to the integration specs and not generally useful to other examples in the test suite?

Also, I'm generally not a fan of defining constants in test suites using `class` or `module`, because they tend to sit in the global namespace and cause potential naming conflicts. You've mitigated this here by putting this in a module that's less likely to conflict, so this point doesn't exactly constitute blocking feedback.

However, I would consider leveraging RSpec's `stub_const`, `Class.new`, or both to create test constants that only survive the scope of an example. This might look like in `helpers.rb`:

```ruby
module ActiveModelSerializersHelpers
  def ams_0_10_or_newer?
    Gem.loaded_specs['active_model_serializers'] \
      && Gem.loaded_specs['active_model_serializers'].version >= Gem::Version.new('0.10')
  end

  def disable_logging
    if ams_0_10_or_newer?
      ActiveModelSerializers.logger.level = Logger::Severity::UNKNOWN
    end
  end
end

RSpec.shared_context 'AMS serializer' do
  extend ActiveModelSerializersHelpers

  let(:serializer_class) do
    stub_const('TestModelSerializer', Class.new(ActiveModel::Serializer) do
      attributes :name
    end)
  end

  if ams_0_10_or_newer?
    let(:model_base_class) do
      Class.new(ActiveModelSerializers::Model) do
        attr_writer :id
      end
    end

    let(:model_class) do
      stub_const('TestModel', Class.new(model_base_class) do
        attributes :name
      end)
    end
  else
    let(:model_base_class) do
      Class.new(ActiveModelSerializers::Model) do
        attr_writer :id

        def initialize(hash = {})
          @attributes = hash
        end

        def read_attribute_for_serialization(name)
          if name == :id || name == 'id'
            object_id
          elsif respond_to?(name)
            send name
          else
            @attributes[name]
          end
        end
      end
    end

    let(:model_class) do
      stub_const('TestModel', Class.new(model_base_class))
    end
  end
end
```

Then in your test file just add:

```ruby
require_relative 'helpers'

RSpec.describe 'ActiveModelSerializers patcher' do
  include_context 'AMS serializer'

  # Perform test here...
end
```